apps/web/components/eventtype/EventTeamTab.tsx

```diff
@@ -310,6 +328,18 @@ const Hosts = ({
     );
   }, [schedulingType, setValue, getValues, submitCount]);
 
+  const updatedHosts = (changedHosts: Host[]) => {
+    return changedHosts.map((newValue) => {
+      const existingHost = getValues("hosts").find((host: Host) => host.userId === newValue.userId);
+      return existingHost
+        ? {
+            ...newValue,
+            scheduleId: existingHost.scheduleId,
+          }
+        : newValue;
+    });
+  };
+
```

Comment:
Yes this is required, else whenever a new host of same type is added , already added (or existing) host will loose its scheduleId.
So this is done, to ensure previously added hosts do no loose scheduleId.
This is because the host is created from list option in `CheckedHostField` component.

Reply:
Makes sense, please add a comment because I couldn't figure out why it was added just by the code.

Reply:
```suggestion
  const updatedHosts = (changedHosts: Host[]) => {
    const existingHosts = getValues("hosts");
    return changedHosts.map((newValue) => {
      const existingHost = existingHosts.find((host: Host) => host.userId === newValue.userId);
      return existingHost
        ? {
            ...newValue,
            scheduleId: existingHost.scheduleId,
          }
        : newValue;
    });
  };

```

Reply:
added comment and updated